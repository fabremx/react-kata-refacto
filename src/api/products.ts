import { Product } from "../models";

export const getProducts = async (): Promise<Product[]> => {
  const response = await fetch(`http://localhost:3000/api/products.json`);
  const { products }: { products: Product[] } = await response.json();
  return products;
};
