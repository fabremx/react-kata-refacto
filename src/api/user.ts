import { User } from "../models";

export const getUser = async (): Promise<User> => {
  const response = await fetch("http://localhost:3000/api/user.json");
  return await response.json();
};
