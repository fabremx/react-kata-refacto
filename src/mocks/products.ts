import { Product } from "../models";

const BASE_PRODUCT = {
  price: 10,
  image: "image 1",
  quantity: 2,
};

export const NEW_PRODUCT: Product = {
  ...BASE_PRODUCT,
  id: 1,
  title: "New Product",
  isNew: true,
  isSoonEnding: false,
};
export const ENDED_PRODUCT: Product = {
  ...BASE_PRODUCT,
  id: 2,
  title: "Ended Product",
  isNew: false,
  isSoonEnding: true,
};
export const COMMON_PRODUCT: Product = {
  ...BASE_PRODUCT,
  id: 3,
  title: "Common Product",
  isNew: false,
  isSoonEnding: false,
};
