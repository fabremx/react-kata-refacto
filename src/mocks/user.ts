import { User } from "../models";

const BASE_USER = {
  id: 1,
  firstname: "firstname",
  lastname: "lastname",
};
export const MOCKED_USER = {
  ...BASE_USER,
  isVIP: false,
  coupon: 0,
};
export const USER_VIP_WITH_COUPON: User = {
  ...BASE_USER,
  isVIP: true,
  coupon: 10,
};
export const USER_VIP_WITHOUT_COUPON: User = {
  ...BASE_USER,
  isVIP: true,
  coupon: 0,
};

export const USER_NOT_VIP_WITH_COUPON: User = {
  ...BASE_USER,
  isVIP: false,
  coupon: 10,
};
export const USER_NOT_VIP_WITHOUT_COUPON: User = {
  ...BASE_USER,
  isVIP: false,
  coupon: 0,
};
