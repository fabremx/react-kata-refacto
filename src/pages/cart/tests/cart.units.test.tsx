import { COMMON_PRODUCT } from "../../../mocks/products";
import { MOCKED_USER } from "../../../mocks/user";
import { mockRetrivedUserAndProductsWith, renderComponent } from "./utils";

jest.mock("../../../components/productTable", () => ({
  ProductTable: () => <>Mock ProductTable</>,
}));

jest.mock("../../../components/summary", () => ({
  Summary: () => <>Mock Summary</>,
}));

describe("Cart", () => {
  it("should render Loading message when user is not defined", async () => {
    // Given
    mockRetrivedUserAndProductsWith(undefined, [COMMON_PRODUCT]);
    // When
    const { asFragment } = await renderComponent();
    // Then
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render component when user is defined", async () => {
    // Given
    mockRetrivedUserAndProductsWith(MOCKED_USER, [COMMON_PRODUCT]);
    // When
    const { asFragment } = await renderComponent();
    // Then
    expect(asFragment()).toMatchSnapshot();
  });
});
