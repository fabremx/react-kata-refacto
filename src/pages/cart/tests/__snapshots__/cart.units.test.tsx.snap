// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should render Loading message when user is not defined 1`] = `
<DocumentFragment>
  <div>
    Loading ...
  </div>
</DocumentFragment>
`;

exports[`Cart should render component when user is defined 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-testid="cart"
  >
    Mock ProductTableMock Summary
  </div>
</DocumentFragment>
`;
